packages/@ember/-internals/utils/lib/invoke.ts

```diff
@@ -46,11 +48,34 @@ export function canInvoke(obj: any | null | undefined, methodName: string): obj
   @param {Array} [args] The arguments to pass to the method
   @return {*} the return value of the invoked method or undefined if it cannot be invoked
   @public
+  @deprecated Use Javascript's optional chaining instead.
 */
 export function tryInvoke(
   obj: any | undefined | null,
   methodName: string,
   args: Array<any | undefined | null>
+) {
+  deprecate(
+    `Use of tryInvoke is deprecated. Instead, consider using JavaScript's optional chaining.`,
+    false,
+    {
+      id: 'ember-utils.try-invoke',
+      until: '4.0.0',
+      for: 'ember-source',
+      since: {
+        available: '3.24.0',
+      },
+      url: 'https://deprecations.emberjs.com/v3.x#toc_ember-utils-try-invoke',
+    }
+  );
+
+  return _tryInvoke(obj, methodName, args);
+}
+
+export function _tryInvoke(
+  obj: any | undefined | null,
+  methodName: string,
+  args: Array<any | undefined | null>
```

Comment:
Let's not split this, we should update internal usages in the same way we expect our users to update.

Reply:
On it. I was aiming at smallest change possible but it makes sense.

Reply:
Ya, I'm worried if even _we_ can't follow our own guidance 😩 😝